src/Orleans/Configuration/ClusterConfiguration.cs

```diff
@@ -441,7 +441,7 @@ internal static IPAddress ResolveIPAddress(string addrOrHost, byte[] subnet, Add
                 }
 
                 var candidates = new List<IPAddress>();
-                IPAddress[] nodeIps = Dns.GetHostAddresses(addrOrHost);
+                IPAddress[] nodeIps = Dns.GetHostAddressesAsync(addrOrHost).GetAwaiter().GetResult();
```

Comment:
Why not simply `Dns.GetHostAddressesAsync(addrOrHost).Result`?

Alternatively, we could make `ResolveIPAddress` async, return a `Task<IPAddress>`, and do `IPAddress[] nodeIps = await Dns.GetHostAddressesAsync(addrOrHost)`. Then the caller would decide on whether to block the thread.